### ISSUE TITLE
Bias in Generator Architecture Choice?

### ISSUE THREAD

**Issue by raunakdoesdev** [Muzammal-Naseer/CDA#4]
@Muzammal-Naseer @salman-h-khan I was reading through your paper and code and saw that "generators are learned against four pre-trained ImageNet models", but also that your black-box cross-domain perturbation was measured on the ImageNet validation set. 

Doesn't training your generator to fool an ImageNet pre-trained discriminator introduce a domain knowledge (from the pretrained discriminator) that no longer renders the target dataset/distribution "unknown"? Let me know if this question is not clear and I can clarify, thank you!

**Comment by Muzammal-Naseer** [comment]
Hi, 

Table 3  in the experimental section shows results where the attacker is unaware of the target a) dataset, b) label space. The target data distribution remains unknown as the generator plays a game with pretrained discriminator (ChexNet, trained on x-ray dataset). We then use these adversarial generators to transfer perturbations to ImageNet model.

We have done other experiments and settings are clearly explained in Section 5.1

Thanks

**Comment by raunakdoesdev** [comment]
Thank you for your rapid response and for pointing out Table 3, in which the discriminator was not pretrained on ImageNet (rather, it was ChexNet trained on x-ray dataset). However, in your Github project, I didn't see this experiment implemented or a pointer to the weights of the pre-trained ChexNet model that you used. Is this information available somewhere? It would help in reproducing the results from your paper.

**Comment by Muzammal-Naseer** [comment]
Hi, 
There are many opersource implementations and this repository (https://github.com/zoogzog/chexnet) provides a pre-trained ChexNet as well.

**Comment by krishnakanthnakka** [comment]
Hi, 

In Table 3 caption, it says 'Adversarial generators are trained against ChexNet on Paintings, Comics and ChestX datasets'.

For example,  I understand 'Our-Comics' attack means generator trained with Comics dataset images against ChexNet discriminator pretrained on its own corresponding COMICS dataset, or is it rather CHESTX dataset? 

Can you please reclarify on which dataset the discriminator is pretrained for each attack in Table 3 during training time? 

Thanks in advance.

**Comment by Muzammal-Naseer** [comment]
Hi,

Table 3 (Training Stage): pretrained Discriminator is trained on ChestXn dataset while generators are trained using comics or paintings, etc. 

Table 3 (Inference Stage): VGG, ResNet-50, Dense121 are pretrained ImageNet models. Test set is ImageNet validation set (50k) images.

I hope this is clear but please feel free to ask any question.
